validator/proposer/service.go

```diff
@@ -68,10 +88,73 @@ func (p *Proposer) Stop() error {
 	return nil
 }
 
+// SaveBlockToMemory will save the block to memory so that the received
+// attestations can be aggregated in it.
+func (p *Proposer) SaveBlockToMemory(block *pbp2p.BeaconBlock) {
```

Comment:
I don't understand why this is done in `proposer` package. This seems more on a beacon node level or validator level scope